src/server/scripts/Kalimdor/HallsOfOrigination/halls_of_origination.cpp

```diff
@@ -0,0 +1,421 @@
+                if (Creature* Isiset = instance->GetCreature(DATA_ISISET))
+                    if (WorldObject* tank = Isiset->AI()->SelectTarget(SELECT_TARGET_TOPAGGRO))
+                        targets.remove(tank);
+
+            targets.remove_if(Trinity::ObjectTypeIdCheck(TYPEID_PLAYER, false));
+            if (targets.empty())
+                return;
+
+            WorldObject* target = Trinity::Containers::SelectRandomContainerElement(targets);
+            targets.clear();
+            targets.push_back(target);
+        }
+
+        void Register() override
+        {
+            OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_hoo_energy_flux_target_selector_SpellScript::FilterTargets, EFFECT_0, TARGET_UNIT_SRC_AREA_ENEMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_hoo_energy_flux_target_selector_SpellScript();
+    }
+};
+
+// 74880 - Arcane Energy
+class spell_hoo_arcane_energy_check : public SpellScriptLoader
+{
+public:
+    spell_hoo_arcane_energy_check() : SpellScriptLoader("spell_hoo_arcane_energy_check") { }
+
+    class spell_hoo_arcane_energy_check_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_hoo_arcane_energy_check_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_ARCANE_BURST, SPELL_ARCANE_ENERGY });
+        }
+
+        void AfterProc(AuraEffect const* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+        {
+            if (GetTarget()->GetPowerPct(POWER_ENERGY) == 100.0f)
+            {
+                GetTarget()->CastSpell((Unit*)nullptr, SPELL_ARCANE_BURST);
+
+                // Stacks should probably be consumed, right? (note: this ability doesn't work on retail)
+                GetTarget()->RemoveAurasDueToSpell(SPELL_ARCANE_ENERGY);
+                GetTarget()->SetPower(POWER_ENERGY, 0);
+            }
+        }
+
+        void Register() override
+        {
+            AfterEffectProc += AuraEffectProcFn(spell_hoo_arcane_energy_check_AuraScript::AfterProc, EFFECT_0, SPELL_AURA_PROC_TRIGGER_SPELL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_hoo_arcane_energy_check_AuraScript();
+    }
+};
+
+// 73686 Fixate
+class spell_hoo_fixate : public SpellScriptLoader
+{
+public:
+    spell_hoo_fixate() : SpellScriptLoader("spell_hoo_fixate") { }
+
+    class spell_hoo_fixate_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_hoo_fixate_SpellScript);
+
+        bool Validate(SpellInfo const* spellInfo) override
+        {
+            if (!spellInfo->GetEffect(EFFECT_0))
+                return false;
+            SpellEffectInfo const* spellEffectInfo = spellInfo->GetEffect(EFFECT_0);
+            return spellEffectInfo && ValidateSpellInfo({ uint32(spellEffectInfo->BasePoints) });
+        }
+
+        void HandleScriptEffect(SpellEffIndex effIndex)
```

Comment:
```cpp
src/server/scripts/Kalimdor/HallsOfOrigination/halls_of_origination.cpp:394:47:
 fatal error: unused parameter 'effIndex' [-Wunused-parameter]
        void HandleScriptEffect(SpellEffIndex effIndex)
                                              ^
```

Reply:
I hope the comment above was a simple enough.
All that is needed is to comment out the parameter.
(feel free to tell me if it should have been more detailed)

Reply:
Yeah it was. Aok told me to finish first part of sql to push it, ill do it now :P